### PR TITLE
🤳🏻 Enable collectibles and NFT as profile picture

### LINKED ISSUE #12613: Enable NFTs on Mobile
The UI for showing collectibles is already merged but hidden behind a feature flag. Enable the UI, allow users to set NFTs as PFP and polish edge cases.

### User Story
As a I should be able to view my collectibles in the wallet.

### PR DESCRIPTION
**Pull request by shivekkhurana**
fixes #12613

### Summary
- Enable NFTs view (based on selected network, only works with mainnet and rinkyby)
- Allow setting NFTs as Profile picture
- View NFT details

- TODO in next PR: Make NFT txns

#### Platforms

- Android
- iOS

#### Areas that maybe impacted
- Profile pictures

##### Functional

- 1-1 chats
- public chats
- group chats
- wallet / transactions
- new account
- user profile updates

status: Ready

### CODE REVIEW

**Comment by shivekkhurana** [comment]
<details>
  <summary>📸 View Screenshot</summary>
  
Simulator Screen Shot - iPhone 11 - 2021-09-27 at 16 40 52
</details>

[🔗 Figma Design](https://www.figma.com/file/XUehMnhyD1FGcWzvGz6SXqvh/Wallet?node-id=11810%3A40829)

**Comment by hesterbruikman** [comment]
@shivekkhurana Could you please make a minor edit to the Collectibles empty state:
It currently says "you will share your **wallet** and IP address", this should be "You will share your **account** and IP address". For consistency. Also, this is my bad, I proposed the copy using 'wallet' earlier.

**Comment by hesterbruikman** [comment]
Also curious about the toast message. What flexibility do we have there when it comes to formatting? e.g. Are emoji's an option?

**Comment by shivekkhurana** [comment]
> Also curious about the toast message. What flexibility do we have there when it comes to formatting? e.g. Are emoji's an option?

Everything is allowed. All React components are valid toasts, however I cannot promise anything until I check. If you want emoji's send me a design (or a rough description) and I'll try it out.

**Comment by hesterbruikman** [comment]
> 
> 
> > Also curious about the toast message. What flexibility do we have there when it comes to formatting? e.g. Are emoji's an option?
> 
> Everything is allowed. All React components are valid toasts, however I cannot promise anything until I check. If you want emoji's send me a design (or a rough description) and I'll try it out.

Awesome! I wasn't thinking of styling the component tbh, but thinking it's a good opportunity for more casual language. We don't use that often, because it's easy to go wrong. But this is a fun low-risk context:

"Profile picture updated😎"

File: src/status_im/ui/components/toastable_highlight.cljs [hunk 1]
```diff
@@ -0,0 +1,116 @@
+(ns status-im.ui.components.toastable-highlight
+  (:require [reagent.core :as reagent]
```
Comment (flexsurfer) [inline src/status_im/ui/components/toastable_highlight.cljs:3]: could you please add a comment what is this component about , thanks

File: src/status_im/ui/screens/wallet/account/views.cljs [hunk 2]
```diff
@@ -115,13 +117,95 @@
                            :border-color     colors/gray-lighter
                            :border-radius    16}}]])])
 
+(defn nft-trait-card [trait]
```
Comment (flexsurfer) [inline src/status_im/ui/screens/wallet/account/views.cljs]: could we move nft related to separate ns ?

**Review by flexsurfer — APPROVED** [review]

**Comment by churik** [comment]
@shivekkhurana 
I can't see any images of collectibles on my Android 10.

IMAGE 2021-09-29 14:23:04

IOS build can't be built.
Can you please take a look?

**Comment by shivekkhurana** [comment]
@churik Do you have an error for ios builds?

**Comment by churik** [comment]
Can be found here: https://ci.status.im/job/status-react/job/prs/job/ios/job/PR-12615/7/console

**Comment by jakubgs** [comment]
No idea why but doing `make nix-update-pods` in a separate PR worked and fixed issues in this one: https://github.com/status-im/status-react/pull/12676

**Comment by churik** [comment]
@shivekkhurana thank you for PR! 

### ISSUE 1: Missing translation in settings
IMAGE 2021-10-01 16:29:07


### ISSUE 2: Still can see amount of collectibles on Ropsten network, but they are not tappable

**Steps:**
1) restore multiaccount with collectible from seed phrase(PR build by default is using Ropsten)
2) Enable Collectibles

**Expected result:**
Collectibles are disabled, it is explained somehow in UI (i.e. `Collectibles are shown only on mainnet`)

**Actual result:**

IMAGE 2021-10-01 16:41:35
**OS:** IOS, Android

### ISSUE 3: Handling of collectibles without images

- [x] 3.1. Set some placeholder for collections without images 
Monosnap 2021-10-01 16-52-20

- [ ] 3.2. Set placeholders for collectibles without placeholders
Monosnap 2021-10-01 16-53-31

- [x] 3.3. Disable `Use as profile photo` for collectibles without images as it can't be set
Monosnap 2021-10-01 16-56-08

### ISSUE 4: Number of collectibles on OpenSea website and in status app is different

**Steps:**
1) add watch-only account to multiaccount (`0xcf2272205cc0cf96Cfbb9Dd740BD681D1E86901E`)
2) count collectibles
3) tap `Open in Opensea` and compare value

**Expected result:**
amount of collectibles should be the same (12)
**Actual result:**
12 in OpenSea, 13 in Status
The redundant is MegaCryptoPolis, it is not tappable from wallet.
**OS:** IOS, Android

### ISSUE 5 (question): why the background color is black when you set collectible as profile photo? 
IMAGE 2021-10-01 17:18:46

**Comment by shivekkhurana** [comment]
## 1 - Fixed ✅ 
## 2 - Fixed ✅ 
## 3 - Added placeholders <strike>Adding an svg renderer 🟡 </strike>
Realised that svgs are not as problematic as empty images or mp4s. Added conditions for them.
Only one item left:
 [ ] 3.2. Set placeholders for collectibles without placeholders  **need design help to get placeholder images**
## 4 - Checking but looks bad - most likely related to APIs 😢 
## 5 - Okay to go with black bg for this release ✅ 
> So background is also visible in chat list and chat. I think it's fine to release as is, background doesn't look too bad and wouldn't appear frequently
Screenshot 2021-10-06 at 1 26 33 PM

**Comment by churik** [comment]
@shivekkhurana thank you for the fixes!

### ISSUE4 
is reproducible; it is not only that particular account, but I also can see differences for another accounts (i.e. simona.eth on opensea has 179 items, in status.im 183 items are shown)

### ISSUE5 No sorting, so no way to find smth you need in random list
Better to sort them alphabetically.

I suppose we can fix separately ISSUE 5, but not ISSUE 4.

File: src/status_im/multiaccounts/core.cljs [hunk 3]
```diff
@@ -198,6 +198,20 @@
...
               (bottom-sheet/hide-bottom-sheet))))
 
+(fx/defn save-profile-picture-from-url
+  {:events [::save-profile-picture-from-url]}
+  [cofx url]
+  (let [key-uid (get-in cofx [:db :multiaccount :key-uid])]
+    (fx/merge cofx
+              {::json-rpc/call [{:method     "multiaccounts_storeIdentityImageFromURL"
+                                 :params     [key-uid url]
+                                 :on-error   (partial prn :----->)
```
Comment (rasom) [inline src/status_im/multiaccounts/core.cljs]: `log/error`/`log/warn`

File: src/status_im/ui/screens/chat/styles/photos.cljs [hunk 4]
```diff
@@ -6,8 +6,8 @@
 (defn radius [size] (/ size 2))
 
 (defn photo-container [size]
-  {:position      :relative
-   :border-radius (radius size)})
+  {:position         :relative
```
Comment (rasom) [inline src/status_im/ui/screens/chat/styles/photos.cljs]: unnecessary change

File: src/status_im/ui/screens/wallet/collectibles/views.cljs [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+            :text-size      :large
+            :accessibility-label
+            (keyword (str "collection-" index))
+            :icon           (if (seq (:image_url collectible))
+                              [wallet.components/token-icon {:style  {:border-radius 40
+                                                                      :overflow      :hidden
+                                                                      :border-width  1
+                                                                      :border-color  colors/gray-lighter}
+                                                             :source {:uri (:image_url collectible)}}]
+                         :main-icons/photo)
```
Comment (rasom) [inline src/status_im/ui/screens/wallet/collectibles/views.cljs]: aligning is a bit off

**Review by rasom — APPROVED** [review]

**Comment by shivekkhurana** [comment]
## ISSUE4 
Anthony and I did some detective work to find what the issue is.

There are two aspects of the OpenSea api:
1- A list of collections
2- Actual assets list (can get all assets or filter by collection id)

The failing collections (MCP in this case) have 0 assets returned although the collection says 1 asset available. So the app stays stuck at loading state (grayish box with no icon). Rainbow works fine because the don't query collections API and simply fetch all assets (ie. fat payload) and then compute collections from all assets.

We face a problem because we fetch assets on demand, and it gets stuck when there are 0 assets in the collection.

- One possible "easy" solution is to show an error when there are 0 assets in a collection **DONE** ✅
- Other possible solution is to fetch all assets upfront but this will hamper UX because we will be loading all assets upfront. **Dropped - Because fat payloads are no good** 🙅🏻

## ISSUE5 No sorting, so no way to find smth you need in random list
Better to sort them alphabetically **DONE** ✅

---

Also addressed comments from Roman. I have not enabled it in release version yet, but I think its good to go in develop. That will allow a broader review.

**Comment by shivekkhurana** [comment]
@qoqobolo Hey, this is ready to be re-re-tested.

**Comment by qoqobolo** [comment]
@shivekkhurana thanks for the fixes and for all this great work!
Checked the latest fixes and basic flows, everything looks good to me.
PR can be merged.
